pkg/cmd/roachtest/tests/online_restore.go

```diff
@@ -415,3 +465,116 @@ func waitForDownloadJob(
 		}
 	}
 }
+
+type restoreHook struct {
+	sp                     restoreSpecs
+	runOnline, runWorkload bool
+
+	preRestoreHook func(ctx context.Context, t test.Test, c cluster.Cluster, db *gosql.DB) error
```

Comment:
i'm not we need all whole `restoreHook` struct yet. all the perf and correctness tests share is `runRestore`. I think the perf tests would be easier to read without the preRestoreHook and wrapupHooks. 

Could you just have a free floating helper along the lines of `runRestore(ctx, sp, t, c,  runOnline, runWorkload)`?

happy to chat about this design in person.

Reply:
Discussed and updated 🙂